parser/src/lexer.rs

```diff
@@ -1397,6 +1461,183 @@ mod tests {
         }
     }
 
+    macro_rules! test_jupyter_magic_line_continuation_eol {
+        ($($name:ident: $eol:expr,)*) => {
+            $(
+            #[test]
+            fn $name() {
+                let source = format!("%matplotlib \\{}  --inline", $eol);
+                let tokens = lex_jupyter_source(&source);
+                assert_eq!(
+                    tokens,
+                    vec![
+                        Tok::MagicCommand { value: "matplotlib   --inline".to_string(), kind: MagicKind::Magic },
+                    ]
+                )
+            }
+            )*
+        };
+    }
+
+    test_jupyter_magic_line_continuation_eol! {
+        test_jupyter_magic_line_continuation_windows_eol: WINDOWS_EOL,
+        test_jupyter_magic_line_continuation_mac_eol: MAC_EOL,
+        test_jupyter_magic_line_continuation_unix_eol: UNIX_EOL,
+    }
+
+    #[test]
+    fn test_empty_jupyter_magic() {
+        let source = "%\n%%\n!\n!!\n?\n??\n/\n,\n;";
+        let tokens = lex_jupyter_source(source);
+        assert_eq!(
+            tokens,
+            vec![
+                Tok::MagicCommand {
+                    value: "".to_string(),
+                    kind: MagicKind::Magic,
+                },
+                #[cfg(feature = "full-lexer")]
+                Tok::NonLogicalNewline,
```

Comment:
I'm not sure if this is the correct way to do this. It seems the tests in CI are not run  using `--all-features` flag.

Reply:
I think the easiest is to remove the feature flag. We always use `full-lexer`.

Reply:
Yes, but the CI doesn't use it:

https://github.com/astral-sh/RustPython-Parser/blob/2d1f69cbb9e331bd7b29253e7f101b47d92bcbff/.github/workflows/ci.yaml#L18

Or, it uses it only for certain steps:

https://github.com/astral-sh/RustPython-Parser/blob/2d1f69cbb9e331bd7b29253e7f101b47d92bcbff/.github/workflows/ci.yaml#L58